src/Microsoft.AspNetCore.Http/HeaderDictionary.cs

```diff
@@ -97,6 +98,34 @@ public HeaderDictionary(int capacity)
             set { this[key] = value; }
         }
 
+        public long? ContentLength
+        {
+            get
+            {
+                long value;
+                var rawValue = this[HeaderNames.ContentLength];
```

Comment:
Yes, this is the same as before. Kestrel and WebListener will provide more optimized versions of IHeaderDictionary. I didn't optimize this one because it's primarily used in tests.